sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/NamedSpiManager.java

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.autoconfigure;
+
+import javax.annotation.Nullable;
+
+interface NamedSpiManager<T> {
```

Comment:
Because this is a private implementation detail I don't think we get much from having this interface. Let's just have NamedSpiManager be the class you defined below, and if needed for convenience it can have an `.empty()` static factory that initializes it with an empty map, we don't need a dedicated implementation / interface to support the empty case.

Reply:
Agreed. This interface isn't adding much value to the implementation, IMO.